internal/file/watch/watch_test.go

```diff
@@ -354,118 +619,107 @@ func Test_watch_Add(t *testing.T) {
 		dirs []string
 	}
 	type fields struct {
-		w        *fsnotify.Watcher
-		eg       errgroup.Group
-		dirs     map[string]struct{}
-		mu       sync.RWMutex
-		onChange func(ctx context.Context, name string) error
-		onCreate func(ctx context.Context, name string) error
-		onRename func(ctx context.Context, name string) error
-		onDelete func(ctx context.Context, name string) error
-		onWrite  func(ctx context.Context, name string) error
-		onChmod  func(ctx context.Context, name string) error
-		onError  func(ctx context.Context, err error) error
+		w    *fsnotify.Watcher
+		dirs map[string]struct{}
 	}
 	type want struct {
-		err error
+		err  error
+		want *watch
 	}
 	type test struct {
 		name       string
 		args       args
 		fields     fields
 		want       want
-		checkFunc  func(want, error) error
-		beforeFunc func(args)
+		checkFunc  func(want, *watch, error) error
+		beforeFunc func(*testing.T, *fields, args)
 		afterFunc  func(args)
 	}
-	defaultCheckFunc := func(w want, err error) error {
+	defaultCheckFunc := func(w want, got *watch, err error) error {
 		if !errors.Is(err, w.err) {
 			return errors.Errorf("got error = %v, want %v", err, w.err)
 		}
+		if got, want := len(got.dirs), len(w.want.dirs); got != want {
+			return errors.Errorf("dirs length = %d, want %d", got, want)
+		}
+		for name := range w.want.dirs {
+			if _, ok := got.dirs[name]; !ok {
+				return errors.Errorf("dirs %s is not exists", name)
+			}
+		}
```

Comment:
can we verify the internal changes of `*fsnotify.Watcher`?

Reply:
I don't think we have a responsibility to know the details beyond.
If it fails, an error will be returned, so it may be enough to make that judgment.
what do you think?

Reply:
I am not sure.
The `w.Add()` function will starting watching the file, and we can't verify if the file is being watching or not... :(
I think the implementation of the library is bad :(